### PR TITLE
Add a few tests for quaternion. (#1384)

### PR DESCRIPTION
**Pull request by npdeep**
### Summary
Add a few more tests for quaternions. Towards https://github.com/Qiskit/qiskit-terra/issues/1384

### Details and comments
test whether quaternions multiply the way they are expected to
- multiplication of quaternions should match the multiplication of corresponding rotating matrices
- multiplication of a quaternion by -1 should not change the rotation matrix.
- `__str__` and ` __repr__` methods should give correct string representations.
- should generate euler angles gracefully in degenerate cases.
- should raise appropriate exceptions for rotations around invalid axis.

### CODE REVIEW

**Comment by nonhermitian** [comment]
It would be nice if these tests could once again be randomized, as they were before @1ucian0 removed them in #1800.

File: test/python/quantum_info/test_quaternions.py [hunk 1]
```diff
@@ -11,14 +11,18 @@
...
+from qiskit.quantum_info.operators.quaternion import quaternion_from_euler, Quaternion
+
 from qiskit.test import QiskitTestCase
 
 
 class TestQuaternions(QiskitTestCase):
     """Tests qiskit.quantum_info.operators.quaternion"""
 
     def setUp(self):
+        self.rndArray = np.array([0.5, 0.8, 0.9, -0.3])
```
Comment (1ucian0) [inline test/python/quantum_info/test_quaternions.py]: The convention we use for multi word variable names is with an underscore. In this case `random_array`.

File: test/python/quantum_info/test_quaternions.py [hunk 2]
```diff
@@ -11,14 +11,18 @@
...
+        self.norm = la.norm(self.rndArray)
+        self.quatUnnormalized = Quaternion(self.rndArray)
```
Comment (1ucian0) [inline test/python/quantum_info/test_quaternions.py]: Item line 23. In this case `quat_unnormalized`.

**Comment by npdeep** [comment]
@1ucian0 could you look at this now?

**Comment by 1ucian0** [comment]
Excellent! Thanks a lot! (I made some small changes to ensure coverage and to remove unnecessary randomness in the result). Welcome @npdeep with your first contribution to qiskit!

**Pull request merged**